src/Engine.zig

```diff
@@ -53,10 +53,28 @@ pub fn setOptions(engine: *Engine, options: structs.Options) !void {
     engine.options = options;
 }
 
+// Signals mach to stop the update loop.
 pub fn setShouldClose(engine: *Engine, value: bool) void {
     engine.internal.setShouldClose(value);
 }
 
+// Signals mach to wait for an event with timeout before calling update()
+// again.  Mach resets to null, so call during each update() if needed.
+//
+// timeout is in seconds (null disables waiting)
+// - pass std.math.floatMax(f64) to wait with no timeout
```

Comment:
Why are we using ``std.math.floatMax(f64)`` instead of just 0 (zero)?

Reply:
I'm thinking of changing the timeout from ?f64 to f64, where <= 0 disables waiting (which is the same as "wait 0 seconds").  Then there are no special cases - if you want an effectively infinite timeout, use floatMax(f64).  That's the reasoning.

Reply:
That could work, I think.